candle-onnx/src/eval.rs

```diff
@@ -1934,6 +1934,137 @@ fn simple_eval_(
+                //    X.shape = [batch_size, seq_length, input_size],
+                //    Y.shape = [batch_size, seq_length, num_directions, hidden_size],
+                //    initial_h.shape = Y_h.shape = [batch_size, num_directions, hidden_size].
+                let layout = get_attr_opt(node, "layout")?.copied().unwrap_or(0);
+                if layout != 0 {
+                    bail!("RNN currently only supports layout == 0");
+                }
+
+                // The input sequences packed (and potentially padded) into one 3-D tensor
+                // with the shape of `[seq_length, batch_size, input_size]`.
+                let x = get(&node.input[0])?;
+                // XXX: depends on layout
+                let (seq_length, batch_size, _) = x.dims3()?;
+                // The weight tensor for the input gate.
+                // Concatenation of `Wi` and `WBi` (if bidirectional).
+                // The tensor has shape `[num_directions, hidden_size, input_size]`.
+                let w = get(&node.input[1])?;
+                // The recurrence weight tensor.
+                // Concatenation of `Ri` and `RBi` (if bidirectional).
+                // This tensor has shape `[num_directions, hidden_size, hidden_size]`.
+                let r = get(&node.input[2])?;
+
+                // The bias tensor for input gate.
+                // Concatenation of `[Wbi, Rbi]` and `[WBbi, RBbi]` (if bidirectional).
+                // This tensor has shape `[num_directions, 2*hidden_size]`.
+                // Optional: If not specified - assumed to be 0.
+                let b_default: Tensor;
+                let b = match get_opt(3) {
+                    Some(n) => n?,
+                    None => {
+                        b_default = Tensor::zeros(
+                            (num_directions, 2 * hidden_size as usize),
+                            DType::F32,
+                            x.device(),
+                        )?;
+                        &b_default
+                    }
+                };
+
+                // Optional tensor specifying lengths of the sequences in a batch.
+                // If not specified - assumed all sequences in the batch to have length `seq_length`.
+                // It has shape `[batch_size]`.
+                let seq_lens_default: Tensor;
+                let seq_lens = match get_opt(4) {
+                    Some(n) => n?,
+                    None => {
+                        seq_lens_default =
+                            Tensor::full(seq_length as i64, (batch_size,), x.device())?;
+                        &seq_lens_default
+                    }
+                };
+                let seq_lens_is_default =
+                    (seq_lens.to_vec1::<i64>()?.iter()).all(|e| *e as usize == seq_length);
+                if !seq_lens_is_default {
+                    bail!("RNN currently only supports default value of seq_lens");
```

Comment:
Perhaps change to something like:

bail!("RNN currently does not support variable-length sequences. All sequences must use the full sequence length of {}", seq_length);